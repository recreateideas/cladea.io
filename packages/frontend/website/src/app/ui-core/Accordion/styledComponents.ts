import styled, { css } from 'styled-components';
import {
    Accordion as MUIAccordion,
    AccordionDetails as MUIAccordionDetails,
    AccordionSummary as MUIAccordionSummary,
} from '@mui/material';

export const Container = styled.div``;
export const Accordion = styled(MUIAccordion)`
    ${(props) => {
        const {
            theme: {
                dsl: { palette },
            },
        } = props;
        return css`
            background-color: ${palette.neutrals.secondaryBg};
            box-shadow: none;
            background-image: none;
        `;
    }}
`;
export const AccordionDetails = styled(MUIAccordionDetails)`
    ${(props) => {
        const {
            theme: {
                dsl: {
                    layout: { padding },
                },
            },
        } = props;
        return css`
            padding: 0px ${padding.small} ${padding.small} ${padding.small};
        `;
    }}
`;
export const AccordionSummary = styled(MUIAccordionSummary)`
    ${(props) => {
        const {
            theme: {
                dsl: {
                    sizes,
                    layout: { padding },
                },
            },
        } = props;
        return css`
            &.Mui-expanded {
                min-height: ${sizes.accordion.medium};
            }
            > div {
                margin: ${padding.small} 0 !important;
            }
            padding: 0px ${padding.small};
        `;
    }}
`;
export const Title = styled.div``;
