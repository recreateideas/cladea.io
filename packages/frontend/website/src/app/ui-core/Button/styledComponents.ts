import styled, { css } from 'styled-components';
import { Button } from '@mui/material';

export const StyledButton = styled(Button)`
    ${(props) => {
        const {
            theme: {
                dsl: { palette },
            },
        } = props;
        return css`
            &.MuiButton-containedPrimary .MuiButton-label {
                color: ${palette.fonts.primaryInverted};
            }
        `;
    }}
`;
