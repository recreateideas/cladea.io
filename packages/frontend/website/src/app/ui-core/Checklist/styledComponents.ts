import styled from 'styled-components';

export const Container = styled.div`
    width: 100%;
    ul {
        background-color: inherit;
        .MuiListItemIcon-root {
            min-width: auto;
        }
        .MuiButtonBase-root {
        }
    }
`;
