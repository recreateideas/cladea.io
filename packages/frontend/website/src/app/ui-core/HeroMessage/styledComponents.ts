import styled, { css } from 'styled-components';

export const Container = styled.div<{ type: 'error' | 'warn' | 'success' }>`
    ${(props) => {
        const {
            theme: {
                dsl: { hexToRgba, styles, layout, palette },
            },
            type,
        } = props;
        const colorMap: {
            [type: string]: {
                color: string;
                backgroundColor: string;
            };
        } = {
            error: {
                color: palette.fonts.error,
                backgroundColor: hexToRgba(palette.secondary.orange[500], 10),
            },
            success: {
                color: palette.fonts.success,
                backgroundColor: hexToRgba(palette.secondary.green[500], 10),
            },
            warn: {
                color: palette.fonts.warn,
                backgroundColor: hexToRgba(palette.secondary.yellow[500], 10),
            },
        };
        return css`
            ${colorMap[type]}
            white-space: pre;
            padding: ${layout.padding.small};
            border-radius: ${styles.borderRadius};
            overflow-x: auto;
            .section {
                display: flex;
                .icon {
                    margin-right: ${layout.margin.small};
                }
            }
        `;
    }}
`;
