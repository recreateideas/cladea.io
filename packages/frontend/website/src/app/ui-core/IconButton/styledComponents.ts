import styled, { css } from 'styled-components';

export const Container = styled.div<{ shape?: 'square'; disabled?: boolean }>`
    ${(props) => {
        const {
            theme: {
                dsl: { palette, styles },
            },
            disabled,
            shape,
        } = props;
        const bordersRadius = shape === 'square' ? styles.borderRadius : undefined;
        return css`
            ${bordersRadius &&
            css`
                .MuiButtonBase-root {
                    border-radius: ${bordersRadius};
                }
            `}
            ${disabled &&
            css`
                &,
                * {
                    cursor: no-drop;
                    svg {
                        fill: ${palette.neutrals.inactive} !important;
                    }
                }
            `}
        `;
    }}
`;
