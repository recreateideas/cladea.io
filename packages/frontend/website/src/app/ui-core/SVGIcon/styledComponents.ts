import styled, { css } from "styled-components";
import { ColorCode } from "src/app/ui-core";

interface ContainerProps {
  rotate?: number;
  fill: string;
  containerSize?: string;
  backgroundFill: string;
  preserveColor?: boolean;
}

export const Container = styled.div<ContainerProps>`
  transition: all 0.2s;
  display: flex;
  justify-content: center;
  align-items: center;
  ${(props) => {
    const { rotate, fill, containerSize, backgroundFill, preserveColor } =
      props;
    return css`
      ${containerSize &&
      css`
        width: ${containerSize};
        height: ${containerSize};
      `}
      ${!preserveColor &&
      css`
        svg path:first-of-type,
        svg rect:not(:last-of-type) {
          fill: ${fill};
        }
      `}
            ${backgroundFill &&
      css`
        svg #background-fill {
          fill: ${backgroundFill};
        }
      `}
            ${rotate && `transform: rotate(${props.rotate}deg);`}
            position: relative;
      transition: all 0.3s;
    `;
  }}
`;

interface ColorMap {
  [type: string]: ColorCode;
}

export const Dot = styled.div<{ colorCode: "error" | "activity" | undefined }>`
  ${(props) => {
    const colorMap: ColorMap = {
      error: props.theme.dsl.palette.fonts.error,
      activity: props.theme.dsl.palette.secondary.neon[500],
    };

    return css`
      position: absolute;
      height: ${props.theme.dsl.layout.spacing.base};
      width: ${props.theme.dsl.layout.spacing.base};
      ${props.colorCode &&
      css`
        background: ${colorMap[props.colorCode]};
      `};
      border-radius: ${props.theme.dsl.layout.spacing.xsmall};
      top: 0;
      right: 0;
    `;
  }}
`;
