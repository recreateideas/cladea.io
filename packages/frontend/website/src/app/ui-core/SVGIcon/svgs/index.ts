import { ReactComponent as lambda } from './ico-aws-lambda.svg';
import { ReactComponent as alert } from './ico-alert.svg';
import { ReactComponent as api } from './ico-api-container.svg';
import { ReactComponent as s3bucket } from './ico-aws-s3-bucket.svg';
import { ReactComponent as kinesisstream } from './ico-aws-kinesis-stream.svg';
import { ReactComponent as sqsqueue } from './ico-aws-sqs.svg';
import { ReactComponent as eventbridge } from './ico-aws-eventbridge.svg';
import { ReactComponent as apigateway } from './ico-aws-apigateway.svg';
import { ReactComponent as rds } from './ico-aws-rds.svg';
import { ReactComponent as logo } from './logo.svg';
interface Svgs {
    [name: string]: any;
}
const svgs: Svgs = {
    lambda,
    alert,
    api,
    s3bucket,
    kinesisstream,
    sqsqueue,
    eventbridge,
    apigateway,
    rds,
    logo,
};

export default svgs;
