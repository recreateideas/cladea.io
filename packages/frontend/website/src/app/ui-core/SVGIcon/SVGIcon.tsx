import { get as getPath } from "lodash";
import { withTheme } from "styled-components";
import svgs from "./svgs";
import { Container, Dot } from "./styledComponents";
import { Theme } from "src/app/ui-core";

export interface SVGIconProps {
  theme: Theme;
  className?: string;
  name?: string;
  size?: string;
  color?: string;
  colorPath?: string;
  fillBackgroundPath?: string;
  onClick?: () => any;
  dotColorCode?: "activity" | "error";
  rotate?: number;
  containerSize?: string;
  fillBackgroundColor?: string;
  preserveColor?: boolean;
}
export const SVGIcon = withTheme((props: SVGIconProps) => {
  const {
    className = "",
    theme,
    name = "alert",
    size: sizeName = "small",
    color: colorProp,
    colorPath,
    fillBackgroundPath,
    onClick,
    dotColorCode,
    containerSize,
    rotate,
    fillBackgroundColor,
    preserveColor,
  } = props;
  // default color for all SVGs has to be fonts.secondary
  const color = colorProp || theme.dsl.palette.fonts.secondary;
  const Icon = svgs[name.toLowerCase()];
  const {
    dsl: {
      sizes: { icons: sizes },
    },
  } = theme;
  const size = sizes[sizeName];
  const fillColor = colorPath ? getPath(theme.dsl.palette, colorPath) : color;
  const fill = preserveColor ? undefined : fillColor;
  const backgroundFill = fillBackgroundPath
    ? getPath(theme.dsl.palette, fillBackgroundPath)
    : fillBackgroundColor;
  return (
    <Container
      className={`svg-icon ${name.toLowerCase()} ${className} `}
      {...{
        onClick,
        fill,
        rotate,
        containerSize: containerSize || size,
        backgroundFill,
        preserveColor,
      }}
    >
      <Icon width={size} height={size} />
      {dotColorCode && <Dot {...{ colorCode: dotColorCode }} />}
    </Container>
  );
});
