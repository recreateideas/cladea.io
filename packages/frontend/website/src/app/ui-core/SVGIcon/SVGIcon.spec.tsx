import { SVGIcon } from './SVGIcon';
import { TestProvider } from 'src/tests';
import { Container, Dot } from './styledComponents';

const { mountWithProviderAndTheme } = TestProvider();
describe('<SVGIcon/>', () => {
    it('should render', () => {
        const root = mountWithProviderAndTheme(<SVGIcon {...{ name: 'lambda' }} />);
        expect(root).toMatchSnapshot();
    });

    it('should use the default name', () => {
        const root = mountWithProviderAndTheme(<SVGIcon />);
        const container = root.find('.svg-icon .alert');
        expect(container.length).toBe(1);
    });

    it('should use the size passed as prop', () => {
        const root = mountWithProviderAndTheme(<SVGIcon {...{ size: 'large' }} />);
        const icon = root.find('ForwardRef(SvgIcoAlert)');
        expect(icon.props()).toHaveProperty('width', window.theme.dsl.sizes.icons.large);
        expect(icon.props()).toHaveProperty('height', window.theme.dsl.sizes.icons.large);
    });

    it('should rotate the icon the same degrees are passed as prop', () => {
        const root = mountWithProviderAndTheme(<SVGIcon {...{ name: 'lambda', rotate: 45 }} />);
        const container = root.find(Container);
        expect(container).toHaveStyleRule('transform', 'rotate(45deg)');
    });

    it('should use the colorPath when colorPath is passed', () => {
        const root = mountWithProviderAndTheme(
            <SVGIcon {...{ name: 'lambda', colorPath: 'primary.ice[500]' }} />
        );
        const container = root.find(Container);
        expect(container.props()).toHaveProperty('fill', window.theme.dsl.palette.primary.ice[500]);
    });

    it('should preserve the color when preserveColor is passed', () => {
        const root = mountWithProviderAndTheme(
            <SVGIcon {...{ name: 'lambda', preserveColor: true }} />
        );
        const container = root.find(Container);
        expect(container.props()).toHaveProperty('fill', undefined);
    });

    it('should set the backgroundFill', () => {
        const root = mountWithProviderAndTheme(
            <SVGIcon {...{ name: 'lambda', fillBackgroundColor: '#abcdef' }} />
        );
        const container = root.find(Container);
        expect(container.props()).toHaveProperty('backgroundFill', '#abcdef');
    });

    it('should set the backgroundFill using backgroundFillPath', () => {
        const root = mountWithProviderAndTheme(
            <SVGIcon {...{ name: 'lambda', fillBackgroundPath: 'primary.ice[500]' }} />
        );
        const container = root.find(Container);
        expect(container.props()).toHaveProperty(
            'backgroundFill',
            window.theme.dsl.palette.primary.ice[500]
        );
    });

    it('should display the dot with the activity color', () => {
        const root = mountWithProviderAndTheme(
            <SVGIcon {...{ name: 'lambda', dotColorCode: 'activity' }} />
        );
        const dot = root.find(Dot);
        expect(dot.length).toBe(1);
        expect(dot.props()).toHaveProperty('colorCode', 'activity');
    });

    it('should display the dot with the error color', () => {
        const root = mountWithProviderAndTheme(
            <SVGIcon {...{ name: 'lambda', dotColorCode: 'error' }} />
        );
        const dot = root.find(Dot);
        expect(dot.length).toBe(1);
        expect(dot.props()).toHaveProperty('colorCode', 'error');
    });
});
