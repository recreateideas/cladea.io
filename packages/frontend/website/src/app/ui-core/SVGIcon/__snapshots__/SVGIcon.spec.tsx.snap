// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SVGIcon/> should render 1`] = `
Array [
  .c0 {
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 20px;
  height: 20px;
  position: relative;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c0 svg path:first-of-type,
.c0 svg rect:not(:last-of-type) {
  fill: #8697BF;
}

<div
    className="c0 svg-icon lambda  "
    fill="#8697BF"
  >
    <svg
      height="20px"
      width="20px"
    >
      ico-aws-lambda.svg
    </svg>
  </div>,
  "",
]
`;
